src/app/components/cat-names/cat-names.component.ts

```diff
@@ -17,9 +18,8 @@ export class CatNamesComponent implements OnInit {
   ngOnInit() {
     this.text.valueChanges
       .pipe(
-        filter(value => value.length > 0),
-        debounceTime(100),
-        mergeMap(value => this.catNamesService.getCatNames(value))
+        debounceTime(500),
```

Comment:
With very poor connection bug with wrong names will be still reproducible.
Hint: what does the mergeMap do?